src/main/java/cam72cam/immersiverailroading/multiblock/SteamGeneratorMultiblock.java

```diff
@@ -0,0 +1,204 @@
+			player.sendMessage(new TextComponentString("Water: " + String.valueOf(10000 - water.fill(new FluidStack(FluidRegistry.WATER, 10000), false)) + " / 10000"));
+			player.sendMessage(new TextComponentString("Water2: " + String.valueOf(10000 - water2.fill(new FluidStack(FluidRegistry.WATER, 10000), false)) + " / 10000"));
+			//player.sendMessage(new TextComponentString("WaterHandler: " + " / " + waterHandler.toString()));
+			player.sendMessage(new TextComponentString("Origin: " + origin));
+			player.sendMessage(new TextComponentString("Water: " + getPos(watercap)));
+			player.sendMessage(new TextComponentString("Water2: " + getPos(watercap2)));
+			player.sendMessage(new TextComponentString("Steam: " + getPos(steamcap)));
+			player.sendMessage(new TextComponentString("Power: " + getPos(power)));*/
+			return  true;
+		}
+
+		@Override
+		public int getInvSize(BlockPos offset) {
+			return 1;
+		}
+
+		private boolean isCenter(BlockPos offset) {
+			return offset.equals(center);
+		}
+
+		@Override
+		public boolean isRender(BlockPos offset) {
+			return isCenter(offset);
+		}
+
+		@Override
+		public void tick(BlockPos offset) {
+			
+			TileMultiblock powerTe = getTile(power);
+			if (powerTe == null) {
+				return;
+			}
+			
+			try {
+				IFluidHandler water = getTile(watercap).getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null);
+				IFluidHandler water2 = getTile(watercap2).getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null);
+				IFluidHandler steam = getTile(steamcap).getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null);
+				
+				int waterFilled = 10000 - water.fill(new FluidStack(FluidRegistry.WATER, 10000), false);
+				int water2Filled = 10000 - water2.fill(new FluidStack(FluidRegistry.WATER, 10000), false);
+				int steamFilled = 10000 - steam.fill(new FluidStack(IRFluids.FLUID_STEAM, 10000), false);
+				
+				
+				//Transfer Water from Water2 FluidHandler to Main Water FluidHandler
+				if(waterFilled < 10000 && water2Filled > 10) {
+					water.fill(water2.drain(10, true), true);
+				}
+				
+				//Convert Water to Steam
+				if(steamFilled < 10000 && waterFilled >= 10 && hasPower()) {
+					IEnergyStorage energy = powerTe.getCapability(CapabilityEnergy.ENERGY, null);
+					energy.extractEnergy(100, false);
+					
+					water.drain(new FluidStack(FluidRegistry.WATER, 10), true);
+					steam.fill(new FluidStack(IRFluids.FLUID_STEAM, 10), true);
+				}
+				
+				//Output Steam
+				BlockPos outputPos = origin.add(output);
+				IFluidHandler output = FluidUtil.getFluidHandler(world, outputPos, EnumFacing.SOUTH);
+				if(output != null && steamFilled >= 10 && output.fill(new FluidStack(IRFluids.FLUID_STEAM, 10), false) >= 10) {
+					output.fill(steam.drain(10, true), true);
+					
+				}
+				
+			} catch (Exception e) {
+				
+			}
+		}
+
+		@Override
+		public boolean canInsertItem(BlockPos offset, int slot, ItemStack stack) {
+			return true;
```

Comment:
So any block in the structure can have items pumped in?  is that right?

Reply:
no, I will fix that.